File: programmers/Lv_2/HwangJunghoon/위장.py
def solution(clothes):
    dic={}
    answer=1
    for i in range(len(clothes)):
        dic[clothes[i][1]]=0
    for j in range(len(clothes)):
        dic[clothes[j][1]]+=1
# 모든 조합의 합 == 개별 가짓수+1 을 다 곱하고 - 1 (-1은 아무것도 안 입는 조합을 뺀 것.)
    for i in dic.values():
        answer*=(i+1)
    return answer-1

File: programmers/Lv_2/JeongJunKwon/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> clothset = new LinkedHashMap<>();
        for(int i = 0; i < clothes.length; i++) {
            if(clothset.get(clothes[i][1]) == null)
                clothset.put(clothes[i][1], 1);
            else
                clothset.put(clothes[i][1], clothset.get(clothes[i][1]) + 1);
        }

        Collection<Integer> combi = clothset.values();
        Iterator<Integer> iter = combi.iterator();
        
        while(iter.hasNext()) {
            answer *= (iter.next() + 1);
        }
        return answer - 1;
    }
}